app/search/page.tsx

```
@@ -0,0 +1,14 @@
+"use client";
+import { useSearchParams } from "next/navigation";
+
+export default function Search() {
+  const sp = useSearchParams();
+  const q = sp.get("q") ?? "";
+  return (
+    <div style={{ padding: 16 }}>
+      <h2>Search</h2>
+      {/* 危険: 直接埋め込み */}
+      <div dangerouslySetInnerHTML={{ __html: q }} />
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)